### ISSUE TITLE
@Door Trigger Causing ReferenceError

### ISSUE THREAD

**Issue by kitt-schlatter** [League-of-Foundry-Developers/fvtt-module-trigger-happy#71]
I updated Trigger Happy today, and now any script using an @Door[] trigger is hitting an exception when Trigger Happy attempts to process it. This will occur when saving a trigger journal entry, loading a scene, setting off the trigger, etc. I've confirmed that no other modules are interacting poorly with Trigger Happy by creating a fresh project who's only installed module is Trigger Happy, and reproducing the error there as well.

The only evidence that anything has gone wrong is the following exception in the console logs:

> Uncaught (in promise) ReferenceError: id is not defined
>     at TriggerHappy._retrieveFromEntity (trigger.js:1355)
>     at TriggerHappy._manageTriggerEvent (trigger.js:552)
>     at TriggerHappy._parseJournal (trigger.js:426)
>     at trigger.js:367
>     at Array.forEach (<anonymous>)
>     at TriggerHappy._parseJournals (trigger.js:367)
>     at Function._call (foundry.js:194)
>     at Function.callAll (foundry.js:153)
>     at ClientDatabaseBackend.callback (foundry.js:8976)
>     at foundry.js:8915

Update: A simple trigger that is reproducing the issue consistently on my end:
> @Door[3850,2650,3950,2650] @ChatMessage[Hello World]

**Comment by tposney** [comment]
Can you add the @door entry that is causing the problem?

**Comment by kitt-schlatter** [comment]
Yeah, it appears to be the mere mention of an @Door that will cause it. I could repro the error referencing a door either by coordinates or by tag. I didn't try all commands extensively, but I've only been able to reproduce with the @Door trigger.

An example of such a trigger:
> @Door[3850,2650,3950,2650] @ChatMessage[Hello World]

**Comment by p4535992** [comment]
are you using the version 0.8.20 ?

**Comment by kitt-schlatter** [comment]
This was with version 0.8.19. At the time, this was the most recent version I could get officially through the foundry module manager. I just now checked, and an update to 0.8.20 was available. That newest version does indeed seem to have resolved the issue!

Not sure if it was just really unfortunate timing on my end around when I upgraded, but the issue is resolved with the most recent version, so I'm closing this card.

Thanks =3
